src/yukon/backend/src/api/controllers/nodes/__init__.py

```diff
@@ -1,129 +1,232 @@
-            self.name = name
-            self.id = id
-            self.health = health
-            self.mode = mode
-            self.uptime = uptime
-            self.vendor_code = vendor_code
-            self.software_version = software_version
-            self.crc = crc
-            self.hardware_version = hardware_version
-            self.uid = uid
-            self.authenticity = authenticity
-
-        def serialise(self) -> Dict[str, Any]:
-            return {
-                'name': self.name,
-                'id': self.id,
-                'health': self.health,
-                'mode': self.mode,
-                'uptime': self.uptime,
-                'vendorCode': self.vendor_code,
-                'softwareVersion': self.software_version,
-                'crc': self.crc,
-                'hardware_version': self.hardware_version,
-                'uid': self.uid,
-                'authenticity': self.authenticity
-            }
-
-    mock_responses = NodeGetDetailsResponse('node_0', nodeId, 'OK', 'OPERATIONAL', 200,
-                                            990, '4.3.2.1', '0xTOOMUCHBEEF', '1.2.3.4', 'my-awesome-uid',
-                                            'I am authentic')
-
-    return jsonify(mock_responses.serialise())
-
-
-@nodes_controller.route('/<int:nodeId>/shutdown', methods=['PUT'])
-async def node_shutdown(nodeId) -> Any:
-    return "", 200
-
-
-@nodes_controller.route('/<int:nodeId>/firmwareupdate', methods=['PUT'])
-async def node_firmware_update(nodeId, param) -> Any:
-    body = await request.get_json()
-    return body['name'], 200
+#!/usr/bin/env python3
+#
+# Copyright (C) 2019  UAVCAN Development Team  <uavcan.org>
+#               2020  dronesolutions.io. All rights reserved.
+# This software is distributed under the terms of the MIT License.
+#
+
+"""
+.. automodule:: nodes
+   :platform: Unix, Windows
+   :synopsis: Runs the Yukon backend
+   :members:
+
+.. moduleauthor:: Theodoros Ntakouris <zarkopafilis@gmail.com>
+.. moduleauthor:: Nuno Marques <nuno.marques@dronesolutions.io>
+"""
+
+import asyncio
+import os
+import pathlib
+import sys
+import tempfile
+from typing import Any, Awaitable, Dict, Tuple
+
+import pyuavcan
+import pyuavcan.transport.udp
+from quart import Blueprint, Response, jsonify, request
+
+dsdl_generated_dir = pathlib.Path(
+    tempfile.gettempdir(), 'dsdl', f'pyuavcan-v{pyuavcan.__version__}')
+dsdl_generated_dir.mkdir(parents=True, exist_ok=True)
+sys.path.insert(0, str(dsdl_generated_dir))
+
+try:
+    import pyuavcan.application             # noqa E402
+    import uavcan.node                      # noqa E402
+    import uavcan.node.Version_1_0          # noqa E402
+    import uavcan.node.Heartbeat_1_0        # noqa E402
+except (ImportError, AttributeError):
+    sys.stderr.write(
+        '\n\n\033[91mFailed to import required DSDL packages. They might not have been generated correctly\n\033[0m')
+    sys.exit(1)
+
+
+class Controller():
```

Comment:
hmm, not sure if `class Controller():` is the same as `class Controller:`. Hopefully it is.